### PR TITLE
try reporting results in CI

### PR DESCRIPTION
**Pull request by jackyzha0**
## Why

- we should be able to see what test failed at a quick glance

## What changed

- add test reporter to CI

## Versioning

- [ ] Breaking protocol change
- [ ] Breaking ts/js API change

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -42,3 +42,9 @@ jobs:
 
       - name: Test
         run: npm test
```
Comment (masad-frost) [inline .github/workflows/ci.yaml:44]: Btw should we add --bail for CI?

File: vitest.config.ts [hunk 2]
```diff
@@ -10,6 +10,10 @@ export default defineConfig({
     sequence: {
       hooks: 'stack',
     },
+    reporters: process.env.GITHUB_ACTIONS
+      ? ['basic', 'github-actions', 'junit']
```
Comment (masad-frost) [inline vitest.config.ts:14]: Interesting they got their own reporter

**Review by masad-frost — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jackyzha0) [reply .github/workflows/ci.yaml:44]: i think having all results is handy and these minutes dont cost us anything? wdyt

Comment (masad-frost) [reply .github/workflows/ci.yaml:44]: Yeah I guess test runs are pretty fast

**Comment by masad-frost** [comment]
I think. You might need a different name for each job or maybe even test run?

**Pull request merged**